### ISSUE TITLE
the tool errors out with "[ERROR] Cannot retrieve koji API version from: . (Reason: Call failed: getAPIVersion)"

### ISSUE THREAD

**Issue by TomasTomecek** [release-engineering/koji-build-finder#79]
After following readme, I read that I should run the tool and it will create a `config.json` which I can edit afterwards. This did not happen instead I got this error:
```
[root@ff8bd657ce47 src]# java -jar target/koji-build-finder-1.0.0-SNAPSHOT-jar-with-dependencies.jar ./docker-image-sha256\:07f9c20362c2e4e2d90e6c2fae822a93d4bbeb379de797dfe8821bff50fd46d8.x86_64.tar.gz
[INFO] koji-build-finder 1.0.0-SNAPSHOT (SHA: fcc7b6e6)
[INFO] Checksum type: md5
[INFO] Loading checksums from file: checksums-md5.json
[ERROR] Cannot retrieve koji API version from: . (Reason: Call failed: getAPIVersion)
com.redhat.red.build.koji.http.error.XmlRpcTransportException: Call failed: getAPIVersion
        at com.redhat.red.build.koji.http.httpclient4.HC4SyncObjectClient.call(HC4SyncObjectClient.java:137)
        at com.redhat.red.build.koji.KojiClient.setup(KojiClient.java:206)
        at com.redhat.red.build.koji.KojiClient.<init>(KojiClient.java:166)
        at com.redhat.red.build.finder.KojiClientSession.<init>(KojiClientSession.java:55)
        at com.redhat.red.build.finder.BuildFinder.main(BuildFinder.java:639)
Caused by: org.apache.http.client.ClientProtocolException: null
        at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:186)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:71)
        at org.commonjava.util.jhttpc.INTERNAL.conn.TrackedHttpClient.doExecute(TrackedHttpClient.java:65)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:71)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:220)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:164)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:139)
        at com.redhat.red.build.koji.http.httpclient4.HC4SyncObjectClient.call(HC4SyncObjectClient.java:128)
        ... 4 common frames omitted
Caused by: org.apache.http.ProtocolException: Target host is not specified
        at org.apache.http.impl.conn.DefaultRoutePlanner.determineRoute(DefaultRoutePlanner.java:70)
        at org.apache.http.impl.client.InternalHttpClient.determineRoute(InternalHttpClient.java:124)
        at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:183)
        ... 11 common frames omitted
[INFO] Loading builds from file: builds.json
[WARN] Could not generate reports since list of builds was empty
```

And no config.json was created.

**Comment by rnc** [comment]
@TomasTomecek I agree, I don't think it should create a config file by default as the mandatory values(the koji hub/web URLs)  have to be filled in by the user.

@dwalluck  I think this should be changed before 1.0.0 release?

**Comment by TomasTomecek** [comment]
Adding a command `koji-build-finder create-default-config` would be a nice fix IMO.

**Comment by dwalluck** [comment]
The software does create a configuration file in `${user.home}/.koji-build-finder/config.json` (the `${user.home}` Java property is equivalent to bash `${HOME}`). The configuration file used to be located  `${user.dir}/config.json` (the`${user.dir}` Java property is equivalent to bash `${PWD}`). But, I think @rnc was against writing anything altogether, so I even considered removing this feature.

What we can do is print an error message and quit before it hits `getAPIVersion()` if no value is specified and/or the URL cannot be resolved.

Unfortunately, I do not think we can fill in these particular values because it would expose an internal URL. This is what is preventing us from writing a reasonable default.

**Comment by rnc** [comment]
@dwalluck Printing an error and quitting if the brew urls haven't been specified is fine as well ; essentially we should make it as easy as possible, as early as possible, to detect it hasn't been fully configured.

**Comment by TomasTomecek** [comment]
@dwalluck thanks! Maybe logging on stdout the path of config it uses would be really helpful.

**Comment by dwalluck** [comment]
@TomasTomecek I just merged #82 if you want to test it.

**Comment by TomasTomecek** [comment]
LGTM!

```
12:53:29.234 [main] DEBUG c.r.red.build.finder.BuildFinder - Configuration does not exist. Implicitly creating with defaults.
```

Thank you!
